partiql-ast/src/test/kotlin/org/partiql/ast/sql/SqlDialectTest.kt

```diff
@@ -599,51 +599,51 @@ class SqlDialectTest {
         @JvmStatic
         fun exprAggCases() = listOf(
             expect("FOO(x)") {
-                exprAgg {
+                exprCall {
                     function = id("FOO")
                     args += exprVar(id("x"), Expr.Var.Scope.DEFAULT)
                 }
             },
             expect("FOO(ALL x)") {
-                exprAgg {
+                exprCall {
                     function = id("FOO")
                     setq = SetQuantifier.ALL
                     args += exprVar(id("x"), Expr.Var.Scope.DEFAULT)
                 }
             },
             expect("FOO(DISTINCT x)") {
-                exprAgg {
+                exprCall {
                     function = id("FOO")
                     setq = SetQuantifier.DISTINCT
                     args += exprVar(id("x"), Expr.Var.Scope.DEFAULT)
                 }
             },
             expect("FOO(x, y)") {
-                exprAgg {
+                exprCall {
                     function = id("FOO")
                     args += exprVar(id("x"), Expr.Var.Scope.DEFAULT)
                     args += exprVar(id("y"), Expr.Var.Scope.DEFAULT)
                 }
             },
             expect("FOO(ALL x, y)") {
-                exprAgg {
+                exprCall {
                     function = id("FOO")
                     setq = SetQuantifier.ALL
                     args += exprVar(id("x"), Expr.Var.Scope.DEFAULT)
                     args += exprVar(id("y"), Expr.Var.Scope.DEFAULT)
                 }
             },
             expect("FOO(DISTINCT x, y)") {
-                exprAgg {
+                exprCall {
                     function = id("FOO")
                     setq = SetQuantifier.DISTINCT
                     args += exprVar(id("x"), Expr.Var.Scope.DEFAULT)
                     args += exprVar(id("y"), Expr.Var.Scope.DEFAULT)
                 }
             },
             expect("COUNT(*)") {
-                exprAgg {
-                    function = id("COUNT_STAR")
+                exprCall {
```

Comment:
(self-review) modeling difference in `partiql-ast` for `COUNT(*)`. Previously it was an `exprAgg` with a function name of `COUNT_STAR` and no arguments. Changed to just be an `exprCall` with function name of `COUNT` and no arguments, which I think is slightly cleaner.

This is how Trino opts to represent `COUNT(*)` in their AST -- https://github.com/trinodb/trino/blob/429f8a8257d38e09b11a69571456c346f973373e/core/trino-parser/src/main/java/io/trino/sql/ExpressionFormatter.java#L436-L438.